### ISSUE TITLE
[New build - DEFAULT] 21w48a: Event Gate Hotkey Improvements

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#4304]
The way Event Gates work when linked to a Yoke (like the old "Hotkey gate") has been improved.
This will allow a more versatile approach to fire weapon systems allowing for smoother integration of upcoming gameplay elements (repair, mining, scavenging).

- It is now possible to use hotkey 1-9 as Fire group: 
You will still use Main Fire or Secondary Fire to shoot but can choose which weapon group will be active by toggling them on/off with their respective hotkey. 
- It is possible to set a Fire Group to be exclusive (ie it will toggle off all other groups when activated)
- It is still possible to fire a weapon with hotkey 1-9.
- New HUD elements have been added for each hotkey (can be toggled off).
- HUD element display name can be customized.
- HUD elements can be hiden or their display order can be tweaked

Other changes:
- A safe zone of 4km around stations was implemented. Pirates and auto-fire turrets wont attack inside the safe zone.

Community Suggestion:
#4296 Material removal will get back to steel

Hotfixes:
#4276 #4288 Surface heat computation has been reversed to the old (wrong) way to ensure blueprint compatibility
#4299 Bevel collisions incorrect
#4286 #4284 Pasted Mechanism top part is orphaned
#4295 Light brick glow is too strong to see actual color
#4237 Exporting blueprint now clears docked entity info
#4299 Repair End Mission message does not display long enough

**Comment by tsunamayo** [comment]
Build is up!

**Comment by commche** [comment]
oh this is very VERY nice!

**Comment by Kadzar** [comment]
Not sure if I'm doing it wrong, but I'm only seeing 1-4 for hotkeys so far.
![20211130222635_1](https://user-images.githubusercontent.com/66035211/144172074-33bf4b80-1cb5-49dc-a529-81cc6167bdcf.jpg)
![20211130222904_1](https://user-images.githubusercontent.com/66035211/144172082-8fd71114-a1d6-4814-b366-d2d03499dbf7.jpg)

**Comment by tsunamayo** [comment]
@Kadzar yeah sorry I havent added them yet, as I cant order them neatly in the event gate without breaking your existing save.

**Comment by tsunamayo** [comment]
@Uncle-Ulty good point, I will update the check to also check the target station range.
